src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -175,6 +175,21 @@ AND NOT EXISTS (SELECT prs FROM p.results prs
             """)
     long countAssessmentsByExerciseIdSubmittedIgnoreTestRunSubmissions(@Param("exerciseId") Long exerciseId);
 
+    @Query("""
+
+            SELECT COUNT(DISTINCT p) FROM StudentParticipation p left join p.submissions s  left join s.results r
+            WHERE p.exercise.id = :exerciseId
+            AND s.id = (select max(id) from p.submissions)
+            AND r.assessor IS NOT NULL
+            AND r.rated = TRUE
+            AND r.submission.submitted = TRUE
+            AND r.completionDate IS NOT NULL
+            AND (p.exercise.dueDate IS NULL OR r.submission.submissionDate <= p.exercise.dueDate)
+            AND NOT EXISTS (select prs from p.results prs where prs.assessor.id = p.student.id)
+            AND :correctionRound = 1L
```

Comment:
I don't understand this statement. can you add a comment to explain it

Reply:
At the moment the 1L is only the placeholder for the subquery that gets the number of results for all submissions. We didn't include it here so we can test all other implemented features, without the query throwing an exception.
We tried different things like e.g. `AND :correctionRound = (select count(s.results) from s)` but get an `org.hibernate.exception.SQLGrammarException: could not extract ResultSet` exception. We are still trying to figure out why it does not work